cupy/core/_routines_indexing.pyx

```diff
@@ -87,21 +89,31 @@ cpdef ndarray _ndarray_argwhere(ndarray self):
             scan_dtype = numpy.int32
         else:
             scan_dtype = numpy_int64
-        scan_index = _math.scan(nonzero, op=_math.scan_op.SCAN_SUM,
-                                dtype=scan_dtype)
+
+        if nonzero.size >= 128 * 1024:
```

Comment:
Could you add a comment here for what leads to this condition --- is it heuristic? or some tuning policy (based on what dtype, what arch, etc)?

Reply:
+1, will be interesting to try this with optuna, maybe we can unlock more performance.

Reply:
I think that the size at which the method used in this PR (incomplete scan) outperforms current method (complete scan) will depend on the data type, GPU type, etc.

Then, why 128 * 1024? Yes, it is kind of heuristic 😅. In many cases, the size of 128 * 1024 is a bit too small to use this method, and there is no performance improvement, but at the same time the performance will not be worse. That's whyl I'm setting the switching size to 128 * 1024, should I change it?

Reply:
For me I am ok if you just leave what you said above as a comment for this line, and note that you tested with float64.